base/applications/cleanmgr/dialog.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * PROJECT:         ReactOS Disk Cleanup
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Dialog Functions
+ * COPYRIGHT:       Copyright 2020 Arnav Bhatt (arnavbhatt288 at gmail dot com)
+ */
+ 
+#include "precomp.h"
+
+BOOL IsSystemDrive;
+DLGHANDLE DialogHandle;
+WCHAR DriveLetter[ARR_MAX_SIZE];
+UINT CleanmgrWindowMsg;
+
+INT_PTR CALLBACK StartDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    HBITMAP BitmapDrive = NULL;
+    
+    switch (message)
+    {    
+        case WM_INITDIALOG:
+        {
+            BitmapDrive = LoadBitmapW(GetModuleHandleW(NULL), MAKEINTRESOURCEW(IDB_DRIVE));
+
+            if (BitmapDrive == NULL)
+            {
+                DPRINT("LoadBitmapW(): Failed to open the bitmap!\n");
+                EndDialog(hwnd, IDCANCEL);
+                return FALSE;
+            }
+
+            InitStartDlg(hwnd, BitmapDrive);
+            return TRUE;
+        }
+
+        case WM_NOTIFY:
+            return ThemeHandler(hwnd, (LPNMCUSTOMDRAW)lParam);
+
+        case WM_THEMECHANGED:
+            InvalidateRect(hwnd, NULL, FALSE);
+            break;
+
+        case WM_MEASUREITEM:
+        {
+            LPMEASUREITEMSTRUCT lpmis = (LPMEASUREITEMSTRUCT)lParam;
+
+            if (lpmis->itemHeight < 18 + 2)
+                lpmis->itemHeight = 18 + 2;
+
+            break;
+        }
+
+        case WM_DRAWITEM:
+        {
+            if (!DrawItemCombobox(lParam))
+            {
+                DPRINT("DrawItemCombobox(): Failed to initialize the ComboBox!\n");
+                EndDialog(hwnd, IDCANCEL);
+            }
+            break;
+        }
+
+        case WM_COMMAND:
+            if (HIWORD(wParam) == CBN_SELCHANGE)
+            {
+                HWND hComboCtrl = GetDlgItem(hwnd, IDC_DRIVE);
+                int ItemIndex = SendMessageW(hComboCtrl, CB_GETCURSEL, 0, 0);
+                StringCbCopyW(DriveLetter, sizeof(DriveLetter), GetProperDriveLetter(hComboCtrl, ItemIndex));
+            }
+
+            switch(LOWORD(wParam))
+            {
+                case IDOK:
+                    DeleteObject(BitmapDrive);
+                    EndDialog(hwnd, IDOK);
+                    break;
+                case IDCANCEL:
+                    EndDialog(hwnd, IDCANCEL);
+                    break;
+            }
+            break;
+
+        case WM_CLOSE:
+            EndDialog(hwnd, IDCANCEL);
+            break;  
+
+        case WM_DESTROY:
+            DeleteObject(BitmapDrive);
```

Comment:
When this message is received, BitmapDrive is **NOT** initialized to something valid.